tests/resolver/test_filtering.py

```diff
@@ -0,0 +1,405 @@
+                vdescs="not a list",  # type: ignore[arg-type]
+                allowed_namespaces=["OmniCorp"],
+            ),
+            maxlen=0,
+        )
+
+    with pytest.raises(ValidationError):
+        deque(
+            filter_variants_by_namespaces(
+                vdescs=["not a `VariantDescription`"],  # type: ignore[list-item]
+                allowed_namespaces=["OmniCorp"],
+            ),
+            maxlen=0,
+        )
+
+
+# ====================== `filter_variants_by_features` ====================== #
+
+
+def test_filter_variants_by_features(vdescs: list[VariantDescription]):
+    vdesc1, vdesc2, _ = vdescs
+
+    vfeat1 = VariantFeature(namespace="OmniCorp", feature="custom_feat")
+    vfeat2 = VariantFeature(namespace="TyrellCorporation", feature="client_id")
+
+    # No feature allowed - should return empty list
+    assert (
+        list(
+            filter_variants_by_features(
+                vdescs=vdescs,
+                allowed_features=[],
+            )
+        )
+        == []
+    )
+
+    # Non existing feature allowed - should return empty list
+    assert (
+        list(
+            filter_variants_by_features(
+                vdescs=vdescs,
+                allowed_features=[
+                    VariantFeature(namespace="UmbrellaCorporation", feature="AI")
+                ],
+            )
+        )
+        == []
+    )
+
+    # Only `vfeat1` allowed - should return `vdesc1`
+    assert list(
+        filter_variants_by_features(
+            vdescs=vdescs,
+            allowed_features=[vfeat1],
+        )
+    ) == [vdesc1]
+
+    # Only `vfeat2` allowed - should return `vdesc2`
+    assert list(
+        filter_variants_by_features(
+            vdescs=vdescs,
+            allowed_features=[vfeat2],
+        )
+    ) == [vdesc2]
+
+    # Both of vfeats - should return all `vdescs`
+    # Note: Order should not matter
+    assert (
+        list(
+            filter_variants_by_features(
+                vdescs=vdescs,
+                allowed_features=[vfeat1, vfeat2],
+            )
+        )
+        == vdescs
+    )
+
+    assert (
+        list(
+            filter_variants_by_features(
+                vdescs=vdescs,
+                allowed_features=[vfeat2, vfeat1],
+            )
+        )
+        == vdescs
+    )
+
+
+def test_filter_variants_by_features_validation_error(
+    vdescs: list[VariantDescription],
+):
+    vfeat = VariantFeature(namespace="namespace", feature="feature")
+
+    with pytest.raises(ValidationError):
+        deque(
+            filter_variants_by_features(
+                vdescs=vdescs,
+                allowed_features="not a list",  # type: ignore[arg-type]
+            ),
+            maxlen=0,
+        )
+
+    with pytest.raises(ValidationError):
+        deque(
+            filter_variants_by_features(
+                vdescs=vdescs,
+                allowed_features=["not a `VariantFeature`"],  # type: ignore[list-item]
```

Comment:
Perhaps also test for passing `VariantProperty` as `allowed_features`, and below for passing `VariantFeature` as `allowed_properties`.

Reply:
Addressed in b4295f8af7ab9da9e7391f01a574d40b7e44f66e